warning: format '%d' expects type 'int', but argument 3 has type 'size_t'

--- qpe.c.orig	Sun Nov 16 09:45:45 2008
+++ qpe.c	Wed May 30 15:24:00 2012
@@ -97,7 +97,7 @@ int qp_encode( char *out, size_t out_size, char *in, s
 				op+= strlen(paragraph);// +3; /** jump the output + =\r\n **/
 				out_remaining-= (strlen(paragraph)); // Was +3, updated to fix Outlook problems
 
-				QPD fprintf(stdout, "Soft break (%d + %d > 76 char) for '%s'\n", current_line_length, charout_size, paragraph);
+				QPD fprintf(stdout, "Soft break (%Zd + %d > 76 char) for '%s'\n", current_line_length, charout_size, paragraph);
 				
 				/** reinitialize the paragraph **/
 				paragraph[0] = '\0';
@@ -108,7 +108,7 @@ int qp_encode( char *out, size_t out_size, char *in, s
 			}
 
 			snprintf(pp, pp_remaining, "%s", charout);
-			QPD fprintf(stdout,"charout='%s', size=%d, pp_remain=%d result='%s'\n", charout, charout_size, pp_remaining, paragraph);
+			QPD fprintf(stdout,"charout='%s', size=%d, pp_remain=%Zd result='%s'\n", charout, charout_size, pp_remaining, paragraph);
 			pp += charout_size;
 			pp_remaining -= charout_size;
 			p++;
@@ -149,13 +149,13 @@ int qp_encode_from_file( char *fname )
 	out_size = in_size *3;
 	in_buffer = malloc( sizeof(char) *in_size +1);
 	if (in_buffer == NULL) {
-		QPD fprintf(stdout,"Error allocating %d bytes for input buffer\n", in_size);
+		QPD fprintf(stdout,"Error allocating %Zd bytes for input buffer\n", in_size);
 		return -1;
 	}
 
 	out_buffer = malloc( sizeof(char) *out_size *3 +1);
 	if (in_buffer == NULL) {
-		QPD fprintf(stdout,"Error allocating %d bytes for output buffer\n", out_size);
+		QPD fprintf(stdout,"Error allocating %Zd bytes for output buffer\n", out_size);
 		return -1;
 	}
 
@@ -169,7 +169,7 @@ int qp_encode_from_file( char *fname )
 	  ** we segfault ;)  **/
 	*(in_buffer +in_size) = '\0';
 
-	QPD fprintf(stdout,"file %s is loaded, size = %d\n", fname, in_size);
+	QPD fprintf(stdout,"file %s is loaded, size = %Zd\n", fname, in_size);
 
 	qp_encode( out_buffer, out_size, in_buffer, in_size );
 
